packages/expo-task-manager/ios/EXTaskManager/EXTaskService.m

```diff
@@ -257,12 +257,15 @@ - (void)setTaskManager:(id<UMTaskManagerInterface>)taskManager
     }
   }
   
-  // Remove events queue for that app.
-  [_eventsQueues removeObjectForKey:appId];
-  
-  if (!isHeadless) {
-    // Maybe update app url in user defaults. It might change only in non-headless mode.
-    [self _maybeUpdateAppUrl:appUrl forAppId:appId];
+  // This can happen when the manifest fails to load.
+  if (appId != nil) {
```

Comment:
Lift this logic as high as possible - make it the caller's responsibility to pass in a non-nil value